pytket/qir/conversion/conversion.py

```diff
@@ -178,14 +178,54 @@ def __init__(
             ),
         )
 
-        # void __quantum__rt__int_record_output(i64)
-        self.record_output = self.module.module.add_external_function(
-            "__quantum__rt__int_record_output",
-            pyqir.FunctionType(
-                pyqir.Type.void(self.module.module.context),
-                [pyqir.IntType(self.module.module.context, qir_int_type)],
-            ),
-        )
+        self.reg_const = {}
+
+        for creg in self.circuit.c_registers:
+            reg_name = creg[0].reg_name
+            self.reg_const[reg_name] = self.module.module.add_byte_string(
+                str.encode(reg_name)
+            )
+
+        if len(self.reg_const) > 0:
+
+            # void __quantum__rt__int_record_output(i64)
+            self.record_output_i64 = self.module.module.add_external_function(
+                "__quantum__rt__int_record_output",
+                pyqir.FunctionType(
+                    pyqir.Type.void(self.module.module.context),
+                    [
+                        pyqir.IntType(self.module.module.context, qir_int_type),
+                        self.reg_const[list(self.reg_const)[0]].type,
+                    ],
+                ),
+            )
+
+            # void __quantum__rt__result_record_output(result)
```

Comment:
Delete commented-out code.

Reply:
done in @cqc-melf
[clean up](https://github.com/CQCL/pytket-qir/pull/38/commits/0c16847c2eb3ff7a2d33a6203c82b39be8c28564)
[0c16847](https://github.com/CQCL/pytket-qir/pull/38/commits/0c16847c2eb3ff7a2d33a6203c82b39be8c28564)